lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -26,47 +27,71 @@ namespace esi {
 using namespace mlir;
 using namespace circt::esi;
 
-namespace {
+/// Figure out the number of bits a type takes on the wire in verilog. Doing it
+/// here is probably a giant hack. TODO: Establish a canonical method to get
+/// this information.
+static size_t getNumBits(Type type) {
+  size_t w = 0;
+  llvm::TypeSwitch<::mlir::Type>(type)
+      .Case<IntegerType>([&](IntegerType t) { w = t.getIntOrFloatBitWidth(); })
+      .Case<circt::rtl::ArrayType>([&](circt::rtl::ArrayType a) {
+        w = a.getSize() * getNumBits(a.getInnerType());
+      });
+  return w;
+}
+
+//===----------------------------------------------------------------------===//
+// Lower to physical conversions and pass.
+//===----------------------------------------------------------------------===//
 
+namespace {
 /// Lower `ChannelBuffer`s, breaking out the various options. For now, just
 /// replace with the specified number of pipeline stages (since that's the only
 /// option).
 struct ChannelBufferLowering : public OpConversionPattern<ChannelBuffer> {
-  ChannelBufferLowering(MLIRContext *ctx) : OpConversionPattern(ctx, 1) {}
+public:
+  using OpConversionPattern::OpConversionPattern;
 
   LogicalResult
   matchAndRewrite(ChannelBuffer buffer, ArrayRef<Value> operands,
-                  ConversionPatternRewriter &rewriter) const final {
```

Comment:
I moved this method to be out-of-line and reduce the # of lines in the anonymous namespace.